### ISSUE TITLE
Inconsistent behaviour of stack

### ISSUE THREAD

**Issue by Hakuyume** [cupy/cupy#793]
I tried cupy 2.1.0.

```python
np.stack((np.array([0, 1]), np.array([2, 3])), axis=1)
# array([[0, 2], [1, 3]])

cupy.stack((cupy.array([0, 1]), cupy.array([2, 3])), axis=1)
# Traceback (most recent call last):
#   File "***/site-packages/cupy/manipulation/join.py", line 126, in stack
#     'axis {} out of bounds [0, {})'.format(axis, x.ndim))
# numpy.core._internal.AxisError: axis 1 out of bounds [0, 1)
```

**Comment by unnonouno** [comment]
It is fixed via #749

**Comment by Hakuyume** [comment]
Thank you! I confirmed this problem is fixed in master branch.
